get_started_with_membrane/03_elements.md

```diff
@@ -0,0 +1,276 @@
+
+```elixir
+defmodule MyElement do
+  use Membrane.Filter
+
+  # Element implementation
+end
+```
+
+The 'Element implementation' consists of defining pads, options and callbacks. Let's find out how to do that.
+
+## Pads
+
+As you already learned, pads allow the creation of the flow of data between elements. Pads, much like contact pads on a printed circuit board, are inputs and outputs of an element and are used to connect the elements with one another.
+Because of that, there are two types of pads: `input` and `output`. It is worth mentioning that `Source` elements may only contain `output` pads, `Sink` elements contain only `input` pads, and `Filter` and `Endpoint` elements can have both of them.
+
+Every pad should define the format of data that it is expecting. This format can be, for example,
+raw audio with a specific sample rate or encoded audio in a given format.
+
+To send data between elements, their pads need to be linked. There are a couple of rules that apply to pad linking:
+
+* One pad of an element can only be linked with one pad from another element.
+  (Dynamic pads can help with that limitation; you'll learn about them in the `pads_and_linking` chapter)
+* Only links between `output` and `input` pads are allowed.
+* Accepted stream formats of pads have to be compatible.
+
+### Defining pads
+
+Pads can be defined using [def_input_pad](https://hexdocs.pm/membrane_core/Membrane.Element.WithInputPads.html#def_input_pad/2) and [def_output_pad](https://hexdocs.pm/membrane_core/Membrane.Element.WithOutputPads.html#def_output_pad/2) macros. They both accept the pad name and the list of properties. The name allows for the identification of the pad. If an element has a single input or output pad, the convention is to name it `input` or `output`, respectively. The pad properties are listed below:
+
+* `accepted_format` - a pattern for a stream format expected on the pad, for example `%Membrane.RawAudio{channels: 2}`. It serves documentation purposes and is validated in runtime.
+* `flow_control` - configures how back pressure should be handled on the pad. You can choose from the following options:
+  * `auto` - Membrane automatically manages the flow control. It works under the assumption that the element does not need to block or slow down the processing rate, it just processes or consumes the stream as it flows. This option is not available for `Source` elements.
+  * `manual` - You need to manually control the flow control by using the `demand` action on `input` pads and implementing the `handle_demand` callback for `output` pads.
+  * `push` - it's a simple mode where an element producing data pushes it right away through the `output` pad. An `input` pad in this mode should be always ready to process that data.
+* `demand_unit` - only for `flow_control` set to `manual`. Either `:bytes` or `:buffers`, specifies what unit will be used to request or receive demands.
```

Comment:
Hmm, should we allow this?

Reply:
I can imagine situation, when it could be useful, for example `File.Source` linked with a filter with auto input pad. Having `demand_unit: :bytes` in combination with `auto_demand_size` option  in filter pad definition would allow you have better control about amount of demanded data